src/data_structures/lazy_segment_tree.rs

```diff
@@ -62,7 +62,7 @@ impl<T: Debug + Default + Ord + Copy + Display + AddAssign + Add<Output = T>> La
         if element_range.start >= query_range.start && element_range.end <= query_range.end {
             return Some(self.tree[idx]);
         }
-        let mid = (element_range.start + element_range.end) / 2;
+        let mid = element_range.start + (element_range.end - element_range.start) / 2;
```

Comment:
```suggestion
        let mid = element_range.start + (element_range.end - element_range.start) / 2; 
```
Just to restart PR checks